src/_pytest/assertion/util.py

```diff
@@ -241,6 +242,15 @@ def _diff_text(left: str, right: str, verbose: int = 0) -> List[str]:
 
     explanation: List[str] = []
 
+    # if both are strings are truthy and normalize to the same string, but we're here,
+    # their byte representation is different. We convert to utf-8 and compare them as bytes.
+
+    if left and right and normalize("NFD", left) == normalize("NFD", right):
+        return [
+            "Strings are different but normalize to the same string. Comparing their utf-8 encoding.",
+            *_compare_eq_sequence(left.encode(), right.encode()),
```

Comment:
Let's use [the `ascii()` builtin function](https://docs.python.org/3/library/functions.html#ascii) rather than encoding - as a bytestring it's often unclear which bytes correspond to characters, or which characters they are!  Using `ascii()` means you can just search for the repr on e.g. Google and discover what e.g. `\u0308` means.

Reply:
Oh, awesome!! Thanks for the pointers!

I understood that in the example we consider just the ä character (`\xe4`) vs. just an a and the dieresis (¨) character split (`\u0308`).

I had not thought about comparing the ascii representations of the strings; that IS def a better approach!

How about now?